src/derivative_wrappers.jl

```diff
@@ -37,19 +37,13 @@ function derivative(f, x::Union{Number,AbstractArray{<:Number}},
 end
 
 jacobian_autodiff(f, x, odefun) = (ForwardDiff.derivative(f,x),1)
-function jacobian_autodiff(f, x::AbstractArray, odefun)
-  if DiffEqBase.has_colorvec(odefun)
-    colorvec = odefun.colorvec
-    sparsity = odefun.jac_prototype
-    jac_prototype = nothing
-  else
-    colorvec = 1:length(x)
-    sparsity = nothing
-    jac_prototype = odefun.jac_prototype
-  end
+function jacobian_autodiff(f, x::AbstractArray, odefun, chunk_size)
+  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec : 1:length(x)
+  sparsity = odefun.sparsity
+  jac_prototype = odefun.jac_prototype
   maxcolor = maximum(colorvec)
-  chunksize = getsize(default_chunk_size(maxcolor))
-  num_of_chunks = Int(ceil(maxcolor / chunksize))
+  chunk_size = chunk_size==0 ? getsize(default_chunk_size(maxcolor)) : chunk_size
+  num_of_chunks = Int(ceil(maxcolor / chunk_size))
```

Comment:
@huanglangwen I'm a bit confused here: shouldn't the chunksize be passed to `forward_color_jacobian`?

Reply:
ah, sorry forgot this one.
https://github.com/JuliaDiff/SparseDiffTools.jl/blob/36978ea873c4a530f1decf01c99c901d33afbcb4/src/differentiation/compute_jacobian_ad.jl#L76

We should instead build a jaccache here like iip one.

Reply:
Problem is that the OOP cache types don't have a field for `jac_cache`, thus as it stands this does not work.  Should this field be added or not? If so, I might need some help getting the plumbing right.

Reply:
with OOP the assumption is that the cache should really just be local or type information to elide allocations.

Reply:
Ok, I hope the implementation as it stands now is good enough.